projects/is-editable-textbox/src/lib/is-editable-textbox.component.ts

```diff
@@ -0,0 +1,79 @@
+import { ChangeDetectorRef, Component, EventEmitter, forwardRef, Input, OnInit, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+@Component({
+  selector: 'is-editable-textbox',
+  templateUrl: 'is-editable-textbox.component.html',
+  styleUrls: ['is-editable-textbox.component.scss'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => IsEditableTextboxComponent),
+    multi: true,
+  }],
+})
+export class IsEditableTextboxComponent implements OnInit, ControlValueAccessor {
+  @Input()
+  autocomplete: any;
+
+  @Input()
+  placeholder = '';
+
+  @Input()
+  edit = false;
+
+  @Output()
+  changed: EventEmitter<any> = new EventEmitter<any>();
+
+  public value = '';
+
+  disabled: boolean;
+
+  // the method set in registerOnChange to emit changes back to the form
+  private _onChangeCallback = (_: any) => { };
+  private _onTouchedCallback = (_: any) => { };
+
+  constructor(private changeDetector: ChangeDetectorRef) { }
+
+  ngOnInit() {
+  }
+
+  toggleEdit() {
+    this.edit = !this.edit;
+  }
+
+  // change events from the input
+  onChange(event: any) {
+    if (event) {
+      this._onChangeCallback(event.target.value);
+    }
+  }
+
+  /**
+   * Implemented as part of ControlValueAccessor.
+   */
+  writeValue(value: string): void {
+    this.value = value;
```

Comment:
I think you'd need to call changeDetector here. Try adding a demo code which will call `control.setValue('') within a setTimeout.

Reply:
You were right, it didn't work when using timeout